src/shared/json-queue.c

```diff
@@ -108,45 +108,56 @@ void jqueue_close(file_queue * queue) {
  * @param queue pointer to the file_queue struct
  * @post The flag variable may be set to CRALERT_READ_FAILED if the read operation got no data.
  * @post The read position is restored if failed to get a JSON object.
- * @retval NULL No data read or could not get a valid JSON object. Pointer to the JSON object otherwise.
+ * @retval NULL No data read or could not get a valid JSON object or read overlong alert. Pointer to the JSON object otherwise.
  */
 cJSON * jqueue_parse_json(file_queue * queue) {
     cJSON * object = NULL;
     char buffer[OS_MAXSTR + 1];
+    int64_t initial_pos;
     int64_t current_pos;
+    int64_t offset;
     const char * jsonErrPtr;
     char * end;
 
-    current_pos = w_ftell(queue->fp);
+    initial_pos = w_ftell(queue->fp);
 
     if (fgets(buffer, OS_MAXSTR + 1, queue->fp)) {
 
         if (end = strchr(buffer, '\n'), end) {
```

Comment:
This line leads to an infinite loop if the input file contains a binary zero:

```
echo -e '{"test": "te\0st"}' >> /var/ossec/logs/alerts/alerts.json
```

<details><summary>Log</summary>

```
2022/05/18 13:00:40 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:41 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:41 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:42 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:42 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:43 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:43 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:44 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:44 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:45 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:45 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:46 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:46 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:47 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:47 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:48 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:48 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:49 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:49 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:50 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:50 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:51 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:51 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:52 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:52 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
2022/05/18 13:00:53 wazuh-integratord[7186] integrator.c:59 at OS_IntegratorD(): DEBUG: jqueue_next()
2022/05/18 13:00:53 wazuh-integratord[7186] json-queue.c:155 at jqueue_parse_json(): DEBUG: Can't read from 'logs/alerts/alerts.json'. Trying again
```

</details>